### DISCUSSION TITLE
"Cannot connect to GPU Backend" -- Colab message

### DISCUSSION THREAD

**Discussion by Robert-Lewis3** [mrdbourke/pytorch-deep-learning#307] [General]
During 03 Pytorch for Computer Vision, when we want to see the time difference of running on GPU vs. CPU, I received a message stating that Google Colab couldn't connect to the GPU backend. So I completed the lesson on the CPU for now. I only use colab for working through this series, so I highly doubt I've used so much compute power that I need to spend $$. This never occurred working through the first 2 sections. Just curious if anyone can explain this message, and ideally help me fix it. 
 
<img width="731" alt="Screenshot 2023-02-18 at 10 40 50 AM" src="https://user-images.githubusercontent.com/53926491/219882924-cb23b715-27fc-4917-9722-ae77c89c48cc.png">

**Comment by mrdbourke** [comment]
Hey @Robert-Lewis3,

I'm not sure of Google Colab's usage limits on the free GPUs.

They are very vague with their limits (I guess this is on purpose).

But in the end, GPUs aren't free so Google must have to put some kinds of limitations on the free tier.

I personally use Google Colab Pro (around $10/month) and have never experienced a lack of GPU.

Otherwise, most of the course should be completable *without* a GPU, it'll just take longer to compute, to fix this, you could:
* Use less data for models (less data = faster compute but likely poorer performance)
* Use smaller model size (similar reason to the above)
* Train for shorter periods of time (similar reason to the above)

Doing this would mean you still get the concepts and code right, you'll just know for better results to scale things up.

**Reply by Robert-Lewis3** [reply]
Thanks for the response! Figured I'll finish things up on the CPU and see if things 'reset.'
